pkg/meta/annotations.go

```diff
@@ -266,6 +285,42 @@ func GetSkipDeploymentCheck(annotations map[string]string) bool {
 	return lookupBoolAnnotation(annotations, SkipDeploymentCheckAnnotation, false /* default value */)
 }
 
+// GetNMASidecarResource is used to retrieve a specific resource for the NMA
+// sidecar. If any parsing error occurs, the default value is returned.
+func GetNMASidecarResource(annotations map[string]string, resourceName corev1.ResourceName) resource.Quantity {
+	annotationName := GenNMASidecarResourceAnnotationName(resourceName)
+	defVal := DefaultSidecarResource[resourceName]
+	quantityStr := lookupStringAnnotation(annotations, annotationName, defVal.String())
+	if quantityStr == "" {
+		return resource.Quantity{}
```

Comment:
Why and we returning defVal here? Are you treating resource-annotation not set different from resource-annotation=""?

Reply:
Yes, if someone set the annotation but leaves it blank, we aren't going to use the default. I'll add a comment.